patches/server/1054-Convert-new-knockback-cause-to-old-knockback-cause-when-calling-EntityKnockbackByEntityEvent.patch

```diff
@@ -0,0 +1,34 @@
+From 3ad01ad20b59b0429e19ff6242454e1970d04c94 Mon Sep 17 00:00:00 2001
+From: LoliColleen <76620594+LoliColleen@users.noreply.github.com>
+Date: Wed, 6 Mar 2024 09:46:49 +0800
+Subject: [PATCH] Added
+
+---
+ .../bukkit/craftbukkit/event/CraftEventFactory.java   | 11 +++++++++++
+ 1 file changed, 11 insertions(+)
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 4c2e81294..e8e2a53f2 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -2086,6 +2086,17 @@ public class CraftEventFactory {
+         return event;
+     }
+ 
++    public static com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause convertNewCauseToOldCause(EntityKnockbackEvent.KnockbackCause knockbackCause) {
++        return switch (knockbackCause) {
++            case DAMAGE -> com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause.DAMAGE;
++            case ENTITY_ATTACK -> com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause.ENTITY_ATTACK;
++            case EXPLOSION -> com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause.EXPLOSION;
++            case SHIELD_BLOCK -> com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause.SHIELD_BLOCK;
++            case SWEEP_ATTACK -> com.destroystokyo.paper.event.entity.EntityKnockbackByEntityEvent.KnockbackCause.SWEEP_ATTACK;
```

Comment:
Considering it's the same as the bukkit cause, can't we just use KnockbackCause#valueOf ?